yarn-project/noir-contracts/src/contracts/schnorr_auth_witness_account_contract/src/main.nr

```diff
@@ -19,23 +68,62 @@ contract SchnorrAuthWitnessAccount {
           payload.serialize(), 
           GENERATOR_INDEX__SIGNATURE_PAYLOAD
         )[0];
-        _inner_is_valid(message_hash, context.this_address());
+        assert(_inner_is_valid(message_hash, context.this_address()));
         payload.execute_calls(&mut context);
     }
 
     #[aztec(private)]
     fn is_valid(
         message_hash: Field
     ) -> Field {
-        _inner_is_valid(message_hash, context.this_address());
-        0xe86ab4ff
+        if (_inner_is_valid(message_hash, context.this_address())){
+            0xe86ab4ff
+        } else {
+            0
+        }
+    }
+
+    #[aztec(public)]
+    fn is_valid_public(
+        message_hash: Field,
+    ) -> Field {
+        let storage = Storage::init(Option::none(), Option::some(&mut context));
+        let value = storage.approved_action.at(message_hash).read();
+        if (value == 1){
+            0xe86ab4ff
+        } else {
+            0
+        }
+    }
+
+    #[aztec(private)]
+    fn set_is_valid_storage(
+        message_hash: Field,
+        value: Field,
+    ) {
+        assert((value == 0) | (value == 1), "value must be a boolean");
+        assert(context.msg_sender() == context.this_address(), "only the owner can set the storage");
+        // assert(_inner_is_valid(message_hash, context.this_address()), "only the owner can set the storage");
+
+        let selector = compute_selector("_set_is_valid_storage(Field,Field)");
+        let _void = context.call_public_function(context.this_address(), selector, [message_hash, value]);
+    }
+
+    #[aztec(public)]
+    internal fn _set_is_valid_storage(
+        message_hash: Field,
+        value: Field,
+    ) {
+        let storage = Storage::init(Option::none(), Option::some(&mut context));
+        storage.approved_action.at(message_hash).write(value);
     }
 
     fn _inner_is_valid(
         message_hash: Field,
         address: Field,
-    ) {
+    ) -> pub bool{
         let witness = get_auth_witness(message_hash);
         assert(recover_address(message_hash, witness) == address);
+        true
```

Comment:
Should not be callable since it is not using `#[aztec(private)]` nr `#[aztec(public)]`. But can see that the abi is read as if it is. Will try executing it to see what actually happens.